### ISSUE TITLE
Add some memory usage info to atm log

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#1611]
We could report memory used (metadata excluded) to store
- fields in the FM
- grid data (dof gids, geometry data)
- atm scratch mem buffer
- I/O host/dev temporaries (for averaging and/or handling padding).

**Comment by PeterCaldwell** [comment]
Yeah, I think this is a good idea. I just spent the last few days thinking my syrah.llnl.gov ne120 run crashed because of the regrid indices bottleneck, but really I think I just didn't use enough nodes... but I'm still not sure because I have no idea how much memory to expect the code to be using.
